apps/hash-graph/lib/graph/src/store/postgres/query/condition.rs

```diff
@@ -89,6 +92,23 @@ impl Transpile for Condition<'_> {
                 fmt.write_str(" && ")?;
                 rhs.transpile(fmt)
             }
+            Condition::StartsWith(lhs, rhs) => {
+                lhs.transpile(fmt)?;
+                fmt.write_str(" LIKE ")?;
+                rhs.transpile(fmt)?;
+                fmt.write_str(" || '%'")
+            }
+            Condition::EndsWith(lhs, rhs) => {
+                lhs.transpile(fmt)?;
+                fmt.write_str(" LIKE '%' || ")?;
+                rhs.transpile(fmt)
+            }
+            Condition::Contains(lhs, rhs) => {
+                lhs.transpile(fmt)?;
+                fmt.write_str(" LIKE '%' || ")?;
+                rhs.transpile(fmt)?;
+                fmt.write_str(" || '%'")
+            }
```

Comment:
We've discussed this offline:
Since we're dealing with `text` here, we wouldn't be able to query into entity properties with this query. We'd have `JSONB` in the field and `TEXT` in the parameter. This causes an exception
```
╰─▶ db error: ERROR: operator does not exist: jsonb ~~ text
    HINT: No operator matches the given name and argument types. You might need to add explicit type casts.
    ├╴at /home/ahmad/dev/hash/hash/apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity/read.rs:84:14
    ├╴backtrace with 215 frames (1)
    ╰╴span trace with 4 frames (1)
```

This is unfortuante as it would have been correct if we could somehow cast the search parameter to JSONB, but we don't have enough information about the type we're operating on to decide to do this or not, just the fact that we're in `properties` for the query path which isn't ideal to base this casting logic on.

Furhtermore we'd want to support lists in the querying and not just strings as mentioned in the above comment - probably not trivial to implement.

Reply:
This wasn't an easy fix but I solved it by converting `jsonb` to `text` on demand.